PublicResolver.lll

```diff
@@ -0,0 +1,246 @@
+;;; ---------------------------------------------------------------------------
+;;; @title A public resolver; only allows the node's owner to set its address.
+;;; @author Daniel Ellison <daniel@syrinx.net>
+
+(seq
+
+  ;; --------------------------------------------------------------------------
+  ;; Constant definitions.
+
+  ;; Booleans.
+  (def 'true 1)
+  (def 'false 0)
+
+  ;; Memory layout.
+  (def 'scratch-one 0x00) ; Scratch space.
+  (def 'scratch-two 0x20) ; More scratch space.
+  (def 'return-code 0x40) ; Boolean returned to calling contract.
+  (def 'return-data 0x60) ; Data returned to calling contract.
+  (def 'call-result 0x80) ; Data returned from called contract.
+  (def 'call-data   0xa0) ; Data provided to called contract.
+
+  ;; Storage layout.
+  (def 'ens-registry 0x00) ; Address of ens registry.
+
+  ;; Precomputed function IDs.
+  (def 'has-record-type    0x41b9dc2b) ; has(bytes32,bytes32)
+  (def 'supports-interface 0x01ffc9a7) ; supportsInterface(bytes4)
+  (def 'get-node-address   0x3b3b57de) ; addr(bytes32)
+  (def 'set-node-address   0xd5fa2b00) ; setAddr(bytes32,address)
+  (def 'get-content-hash   0xd8389dc5) ; hash(bytes32)
+  (def 'set-content-hash   0xd7fa1007) ; setHash(bytes32,bytes32)
+  (def 'get-node-owner     0x02571be3) ; owner(bytes32)
+
+  ;; Jumping here causes an EVM error.
+  (def 'invalid-location 0x02)  ; Jumping here causes an EVM panic.
```

Comment:
`panic` is currently broken because of recent changes in the assembler to support first-class functions (iirc). Once that's fixed I'll switch this to use `panic`. In the meantime, `(jump 0x02)` does essentially the same thing.